### PR TITLE
Fix unit tests

### PR DESCRIPTION
**Pull request by nschimme**

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -20,7 +20,7 @@ set_target_properties(
   CXX_STANDARD_REQUIRED ON
   CXX_EXTENSIONS OFF
   COMPILE_FLAGS "${WARNING_FLAGS}"
-  UNITY_BUILD ON
+  UNITY_BUILD ${USE_UNITY_BUILD}
```
Comment (mdrcode) [inline tests/CMakeLists.txt:23]: Just curious - is there no way to set all these target_properties in a common location and then inherit? Not a huge deal of course just learning.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -189,13 +189,14 @@ set(adventure_SRCS
         )
 set(TestAdventure_SRCS testadventure.cpp testadventure.h)
 add_executable(TestAdventure ${TestAdventure_SRCS} ${adventure_SRCS})
+add_dependencies(TestAdventure glm)
```
Comment (mdrcode) [inline tests/CMakeLists.txt:192]: So just to double check, *all* tests should add this glm dependency explicitly?

**Review by mdrcode — APPROVED** [review]
LGTM

File: tests/CMakeLists.txt (hunk 2 above)
Comment (nschimme) [reply tests/CMakeLists.txt:192]: GmcpMessage seems to be dependent on src/global/bits.h

We should probably convert our core business logic into a lib that is used by the executable and tests

File: tests/CMakeLists.txt (hunk 1 above)
Comment (nschimme) [reply tests/CMakeLists.txt:23]: I dug through the documentation and couldn't find a way to set it globally

**Pull request merged**